bentoml/_internal/models/store.py

```diff
@@ -119,35 +120,33 @@ def register_model(
             ctx.metadata["params_a"] = value_a
         """
         model_tag = ModelTag.from_str(name)
-        model_path = self._create_model_path(model_tag)
-        model_yaml = Path(model_path, f"{MODEL_YAML_NAMESPACE}{Extensions.YAML}")
+        model_path = self._create_path(model_tag)
+        model_yaml = Path(model_path, f"{MODEL_YAML_NAMESPACE}{YAML_EXT}")
 
         ctx = StoreCtx(
             name=model_tag.name,
+            path=model_path,
             version=model_tag.version,
             labels=labels,
             metadata=metadata,
             options=options,
         )
         try:
-            dump_model_yaml(model_yaml, ctx, framework_context=framework_context)
             yield ctx
+        except StopIteration:
```

Comment:
what's the reason for `StopIteration` here? maybe just `except Exception`?

Reply:
This is mainly for catching the ctx when saving is failed.